File: 6_write_txt/write_txt.py
# coding: utf8
# Python implementation of the write_txt code for Rosetta Stone
# Author: Eric Alcaide

def read_txt(name="data.txt"):
	# Automatically closes the file once outside of it
	with open(name, 'r') as f:
		r = f.readlines()
		first,last = r[0], r[-1]

	return first,last

def write_txt(first, second="Bye bye .txt file"):
    with open('write_py.txt', 'w') as f:
    	f.write(first)
    	f.write('\n'+second)

def main():
	first, second = read_txt()
	write_txt(first,second)

if __name__ == "__main__":
    main()

File: 6_write_txt/write_txt.go
package main 

import (
	"fmt"
	"io/ioutil"
	"strings"
	"os"
)

func read_txt(name string) (string,string){ // 
	b, err := ioutil.ReadFile(name) // just pass the file name
    if err != nil {
        fmt.Print(err)
    }
    str := string(b) // convert content to a 'string'
    chunks := strings.Split(str, "\n")

    first,last := chunks[0], chunks[len(chunks)-1]

    return first,last
}

func write_txt(first string, second string) bool{
	file, err := os.Create("write_go.txt")
    if err != nil {
        fmt.Print(err)
    }
    defer file.Close()

    fmt.Fprintln(file, first)
    fmt.Fprintf(file, "\n")
    fmt.Fprintf(file, second)

    return true
}

func main() {
	first,second := read_txt("data.txt")
	write_txt(first,second)
	fmt.Println("Success.")
}
